src/language-server/definition-finder.ts

```diff
@@ -109,4 +128,39 @@ export default class DefinitionFinder extends Handler {
 
     return locations;
   }
+
+  private findSymbols(queryable: Queryable): SymbolInformation[] {
+    // Most kinds of symbols are covered by JS and TS analyzers. We can
+    // help track down custom elements by tag name though.
+    const symbols: SymbolInformation[] = [];
+    const elements =
+        queryable.getFeatures({kind: 'element', externalPackages: true});
+    for (const element of elements) {
+      if (!element.sourceRange) {
+        continue;
+      }
+      if (element.tagName) {
```

Comment:
not clear when reading the code if this check is here to make typescript typecheck happy or because there are legit cases where sourceRange/tagName can be null. If it's not the latter, I'd suggest to use `element.tagName!`

Reply:
There are some super weird cases where we have something that's an element but we either don't know where exactly it's defined, or what its tag name is.

Think stuff defined in Polymer 1.0 core, or stuff defined only in comments.

Reply:
oh, I see...Yeah elements defined in comments is something i definitely wasn't thinking about!